en/regular-expression.md

```diff
@@ -34,17 +34,142 @@ In cases where we want to have a forward slash to be a part of a regular express
 The different modifiers that are used to perform case-insensitive searches are:
 
 * `g` - global search (finds all matches instead of stopping after the first match)
+
+Example :
+
+```javascript
+const str = "Hello world, hello again!";
+const regex = /hello/g;
+const matches = str.match(regex);
+// If you are thinking about .match() Read This 👇
+// It is a built-in method in JavaScript that is used to search a string for a match against an expression.
+// If the match is found, It returns an array of all the matches that were found. and if not, the .match() method returns null.
+
+console.log(matches); // ["Hello", "hello"]
+```
 * `i` - case insensitive search
-* `m` - multiline matching
+
+Example :
+
+```javascript
+const str = "HeLlO WoRlD";
+const regex = /hello/i;
+const match = regex.test(str);
+// The .test() method returns a boolean value: 
+// true if the match is found, and false if the match is not found.
+
+console.log(match); // true
+```
+
+* `m` - multiline matching 
+
+Example:
```

Comment:
Is it possible to put a space after the Example? Like you have done to the rest?

Reply:
@sumn2u in multiline matching ?

Reply:
<img width="541" alt="image" src="https://github.com/sumn2u/learn-javascript/assets/114233864/2f01941b-f609-4927-a70e-1ef10ad205e7">

is this what you looking for ??

a space between examples ?? @sumn2u

Reply:
or should we go for this 

<img width="634" alt="image" src="https://github.com/sumn2u/learn-javascript/assets/114233864/1e465e53-0e5e-49c7-a84d-296be9efa675">